src/models/category.rs

```diff
@@ -0,0 +1,20 @@
+use crate::models::UserRecordId;
+use serde::{Deserialize, Serialize};
+use surrealdb::sql::Thing;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Category {
+    pub id: Option<Thing>,
+    pub owner: Thing,
+    pub name: String,
+
+    pub created_at: chrono::NaiveDateTime,
+    pub updated_at: chrono::NaiveDateTime,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(crate = "rocket::serde")]
+pub struct CreateCategory {
+    pub name: String,
```

Comment:
```suggestion
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(crate = "rocket::serde")]
pub struct CreateCategory<'d> {
    pub name: &'d str,
```

`name` always a short string, so use borrowed str will improve our performance, otherwise, using `String` will cause `rocket` copy this field multiple times.